bff/pkg/server/server.go

```diff
@@ -513,6 +522,34 @@ func activity_handler(metrics_service metrics.Service, validate *validator.Valid
 			return
 		}
 
+		// populate query_fp with query text
+		for _, result := range results {
+			switch metric := result["metric"].(type) {
+			case map[string]interface{}:
+				if queryFP, ok := metric[query_fp_label].(string); ok {
+					queryText, err := query_storage.GetQuery(queryFP)
+					if err != nil {
+						queryText = "<not found>"
+						log.Printf("Error fetching query text: %s", err)
+					}
+					metric[query_fp_label] = queryText
+					result["metric"] = metric
+				}
+			case map[string]string:
+				if queryFP, ok := metric[query_fp_label]; ok {
+					queryText, err := query_storage.GetQuery(queryFP)
+					if err != nil {
+						http.Error(w, "Error fetching query text: "+err.Error(), http.StatusInternalServerError)
+						return
+					}
+					metric[query_fp_label] = queryText
+					result["metric"] = metric
+				}
```

Comment:
This piece of code should be the same as https://github.com/crystaldb/autodba/pull/93/files#diff-499c853237746b9bdecff8296d4104ae0a884f344b9c367416fa42779acffb1dR529-R537. Let's add a helper function to avoid code duplication.